templates/types/extractor/fastapi/app/services/extractor.py

```diff
@@ -0,0 +1,17 @@
+from app.engine import get_query_engine
+from app.models.request import RequestData
+from app.services.schema import SchemaService
+
+
+class ExtractorService:
+
+    @classmethod
+    async def extract(cls, data: RequestData):
```

Comment:
this service method has just two parameters, probably doesn't need a parameter object (here `RequestData`), instead just use:
```suggestion
    async def extract(cls, query: str, json_schema: str):
```
and move `RequestData` to API layer.

after this you should be only need to update `ExtractorService` once llamaindex supports pydantics v2